### ISSUE TITLE
What is the initial reward ?

### ISSUE THREAD

**Issue by manuel-delverme** [werner-duvaud/muzero-general#21]
https://github.com/werner-duvaud/muzero-general/blob/ecca75c8d5893048b0acc6c5897a504c6334b871/models.py#L137

Since you are predicting a distributions, the first reward should be scalar_to_support(0)

**Comment by werner-duvaud** [comment]
Hi,

I think you did not quite understand the line you mentioned. It returns a zeros tensor of shape batch_size x support_size. Each element is therefore a distribution of zeros corresponding to `scalar_to_support(0)`.

**Comment by manuel-delverme** [comment]
I'm pretty sure I understood this right.

You are predicting the reward as a categorical distribution.

scalar_to_support(0) ought to be [0..1..0], with 1 for the central element.
That code ends up working anyway because in this case:
`x_low * p_low + x_high * p_high` becomes `0 * 1 + 1 * 0`
and it would be equal to torch.zeros(..) but it's conceptually wrong.

It might also cause errors with the loss because you have element loss calculation, if you have: predicted = scalar_to_support(0)
target = torch.zeros(..)
the error is not 0 but 1/support in the MSE case

**Comment by werner-duvaud** [comment]
Okay, sorry I didn't understand what you meant. You are right the distribution should theoretically contain a 1 in the middle.

However, this distribution (as a prediction of the reward for the initial observation) is never really used, it is there only for the consistency of the batch. It is never used to calculate the loss.

But we will add a 1 in the next commit, so that it does not lead to confusion.
Thank you !
